src/forms/Slushbucket.tsx

```diff
@@ -0,0 +1,237 @@
+          }}
+        >
+          <input
+            tabIndex={-1}
+            type="text"
+            autoComplete="false"
+            className={
+              formStyles.FormControl +
+              " " +
+              (this.props.readonly ? inputStyles.ReadOnly : "")
+            }
+            placeholder={placeholder}
+            onChange={e => this.handleInput(e)}
+            onKeyUp={event => {
+              this.handleKeyUp(event);
+            }}
+            value={this.state.searchString}
+            disabled={this.props.readonly}
+            readOnly={this.props.readonly}
+          />
+          <div
+            className={`${styles.Spinner} ${isFetching
+              ? displayStyles.Block
+              : displayStyles.None}`}
+          >
+            <MDSpinner size={18} />
+          </div>
+
+          <div
+            style={{
+              transform: "translateX(-33px)",
+              display: "flex",
+              alignItems: "center",
+              visibility: this.state.searchString === "" ? "hidden" : "visible"
+            }}
+          >
+            <ClearButton
+              onClick={() => {
+                this.resetSearchString();
+              }}
+            />
+          </div>
+        </div>
+
+        <div
+          style={{
+            display: "flex"
+          }}
+        >
+          <div
+            className={styles.Results}
+            style={{
+              marginRight: "20px"
+            }}
+          >
+            <div className={`${styles.SelectedRow}`}>
+              <b>Available</b>
+            </div>
+            <Scrollbars autoHeight autoHeightMin={400} autoHeightMax={400}>
+              {choices
+                .filter(choiceItem => {
+                  if (this.props.readonly) {
+                    return false;
+                  }
+                  if (this.state.searchString === "") {
+                    // if nothing is typed show all results
+                    return true;
+                  } else {
+                    // if user typed search string only show those that contain string
+                    // TODO sort by search string ?
+                    return choiceItem
+                      .display
+                      .toLowerCase()
+                      .includes(this.state.searchString.toLowerCase());
+                  }
+                })
+                .map((choiceItem, i) => {
+                  return (
+                    <div
+                      tabIndex={i + 1}
+                      key={choiceItem.value + '_' + i}
+                      className={`${styles.ResultRow} ${selected.includes(
+                        choiceItem.value
+                      )
+                        ? styles.Active
+                        : ""}`}
+                      onMouseDown={() => {
+                        if (
+                          selected.filter(e => e === choiceItem.value).length === 0
```

Comment:
'e' is a bit of a confusing variable name, makes the function looks like an event handler. Maybe split this (look up a value in the list of selected values) into a helper function?

Reply:
yep a bad habit.
e stands for element but is indeed confusing.
changed it